src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,74 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
 import { Loader } from '../Loader';
+import { AppDispatch } from '../../app/store';
+import { useDispatch } from 'react-redux';
+import { setCurrentTodo } from '../../features/currentTodo';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+  const dispatch: AppDispatch = useDispatch();
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getUser(todo.userId)
+      .then(setUser)
+      .catch(e => {
+        throw new Error(e);
```

Comment:
Throwing an error inside a catch block without handling it can cause the application to crash. Consider logging the error or displaying a user-friendly message instead.